### PR TITLE
SONARPY-2310: Partly support wildcard import

### PR DESCRIPTION
**Pull request by joke1196**
[SONARPY-2310](https://sonarsource.atlassian.net/browse/SONARPY-2310)

[SONARPY-2310]: https://sonarsource.atlassian.net/browse/SONARPY-2310?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: python-frontend/src/main/java/org/sonar/python/semantic/v2/TypeInferenceV2.java [hunk 1]
```diff
@@ -61,10 +62,15 @@ public TypeInferenceV2(TypeTable projectLevelTypeTable, PythonFile pythonFile, S
   }
 
   public Map<SymbolV2, Set<PythonType>> inferTypes(FileInput fileInput) {
-    TrivialTypeInferenceVisitor trivialTypeInferenceVisitor = new TrivialTypeInferenceVisitor(projectLevelTypeTable, pythonFile, fullyQualifiedModuleName);
+    WildcardImportVisitor wildcardImportVisitor = new WildcardImportVisitor(projectLevelTypeTable);
```
Comment (guillaume-dequenne-sonarsource) [inline python-frontend/src/main/java/org/sonar/python/semantic/v2/TypeInferenceV2.java:65]: Why would this need to be its dedicated visitor?
I know we're going for a naive implementation, but I think it's still sensible to assume that we don't want the following resolution:
```
foo()
from lib import * # contains foo
```

I'm under the impression that relying on the main visit of `TrivialTypeInferenceVisitor` should be enough here, no?

Comment (Seppli11) [reply python-frontend/src/main/java/org/sonar/python/semantic/v2/TypeInferenceV2.java:65]: The `TrivialTypeInferenceVisistor` is enough (initially that's what we did). Our thought was that if it is moved to its own visitor, it is easier to rip it out when we eventually will do that. But we didn't think of the example you provided

File: python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java [hunk 2]
```diff
@@ -3124,4 +3262,25 @@ private static Statement lastStatement(StatementList statementList) {
     List<Statement> statements = statementList.statements();
     return statements.get(statements.size() - 1);
   }
+
+  private static class TestProject {
```
Comment (guillaume-dequenne-sonarsource) [inline python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java:3272]: That's a pretty neat construct!
I'm wondering if it wouldn't make more sense for this to be in a different test category though (either `ProjectLevelSymbolTableTest` or `ProjectLevelTypeTableTest`, or even a new `CrossFileTypeInferenceTest` dedicated to tests focused on import resolution.
We probably can keep this here for now though.

**Review by guillaume-dequenne-sonarsource — APPROVED** [review]
Kudos on the clear tests (well documenting the limitations) and the clear separation of commits (even if I didn't notice it initially, sorry about that!).
LGTM.

**Pull request merged**